### PR TITLE
lib/core/stream: add an optional `max` parameter to `read_line`

### PR DESCRIPTION
**Pull request by privat**

### CODE REVIEW

**Comment by privat** [comment]
This PR does not work because `BufferedReader::read_char` is UFT-8 broken and the default `Reader::append_line_to` depends on `read_char`.

**Comment by R4PaSs** [comment]
@privat now that `read_char` supports codecs properly, this could be reopened if still valuable

**Comment by privat** [comment]
reroll

File: lib/core/stream.nit [hunk 1]
```diff
@@ -385,8 +394,20 @@ abstract class Reader
 	# assert i.eof
 	# ~~~
 	#
+	# when `max` is given it is the maximum of characters to read.
```
Comment (R4PaSs) [inline lib/core/stream.nit]: s/maximum of characters/maximum number of characters

**Pull request closed without merging**